CMakeLists.txt

```diff
@@ -79,27 +81,15 @@ ENDIF (CMAKE_SYSTEM_NAME MATCHES "Linux")
 
 SET(CONFIG_HEADER "${PROJECT_SOURCE_DIR}/config.h")
```

Comment:
All of this CONFIG_HEADER stuff really should not exist.  CMake templates should be used if you need something like this, but the define's I'm not sure why they exist, why do you need to do things like `#define CSS_STD 14` in the source, source code should not depend on C++ versions like this, rather feature tests should be done and explicit features should be enabled or disabled, those saved for install via a cmake template (if necessary at all, if not header-only that is).

Reply:
I've removed the CONFIG_HEADER from trantor.

I don't quite know what the `CMake templates` you mentioned is. Can you give me some links to documents or examples? I think the same problem in drogon needs to be solved.

Reply:
CMake templates are very much like Autoconf templates (if you've used the old Makefiles build ecosystem), but instead of calling `AC_CONFIG_HEADERS` like in Autoconf, you instead call [`configure_file`](https://cmake.org/cmake/help/v3.15/command/configure_file.html) (link to the docs).

In essence, you will make a Template `config.h.in` file, with content like:
```cpp
#cmakedefine OpenSSL_FOUND
#cmakedefine OpenSSL_VERSION "@VAR2@"
#cmakedefine01 SOME_FLAG
```
If you do something like this in cmake:
```cmake
if(OpenSSL_FOUND)
  set(OpenSSL_VERSION "Blah")
endif()
option(SOME_FLAG "Set some flag" ON)
configure_file(config.h.in config.h @ONLY)
include_directories(${CMAKE_CURRENT_BINARY_DIR})
```
Will generate something like this:
```cpp
#define OpenSSL_FOUND
#define OpenSSL_VERSION "Blah"
#define SOME_FLAG 1
```
And if OpenSSL_FOUND is false and some_flag is off instead of on, then it would generate this:
```cpp
/* #undef OpenSSL_FOUND */
/* #undef OpenSSL_VERSION */
#define SOME_FLAG 0
```
It has a fairly simple set of primitive instructions (see docs) but it is actually quit powerful and perfect for all config-file needs, not just `#define`'ing as above but variable expansion and replacement and code-gen and more.  :-)

The purpose of it is to keep source code out of your build files.  You should never have source code in build files.  :-)

Reply:
For note, you still don't need to make a config file just for something like `USE_OPENSSL`, the more usual method is to, say you have an SSLContext file in it, the entire API is exposed in the header, the `cpp` file for it contains the implementation, the general way would be to use a different cpp implementation file for if OpenSSL_FOUND compared to if it's not (where the SSLContext would be shimmed out).  This keeps the API identical regardless of how it is compiled (you of course want to throw an error if they try to turn on SSL without it existing in the shim).

Reply:
OvermindDL1 Yes, I see now. I just removed the USE_OPENSSL macro out of the public API header files.

Reply:
@OvermindDL1 And I will take some time to learn the CMake templates. Thank you for your explanation.